[ADD] add module stock_backorder_strategy

laetitia-gangloff opened the pull request:
Define on stock_picking_type how to manage backorder.
create : default behavior
no_create : no backorder created
cancel : backorder is cancelled

pedrobaeza commented on stock_backorder_strategy/README.rst:
```diff
@@ -0,0 +1,60 @@
+==================
+
+In types of operation add a backorder strategy to be able to cancel created
+backorder or to avoid to create a backorder.
```
Please explain here the use cases for each backorder strategy

pedrobaeza commented on stock_backorder_strategy/README.rst:
```diff
@@ -0,0 +1,60 @@
+Configuration
+=============
+
+To configure this module, you need to:
```
Put here the menu path for setting this.

pedrobaeza commented on stock_backorder_strategy/__openerp__.py:
```diff
@@ -0,0 +1,36 @@
+##############################################################################
+
+{
+    "name": "Stock no backorder",
```
Picking backordering strategies

pedrobaeza commented on stock_backorder_strategy/__openerp__.py:
```diff
@@ -0,0 +1,36 @@
+{
+    "name": "Stock no backorder",
+    "version": "0.1",
+    'author': "Acsone, Odoo Community Association (OCA)",
```
You usually put "ACSONE A/V" as the author

pedrobaeza commented on stock_backorder_strategy/models/stock.py:
```diff
@@ -0,0 +1,50 @@
+        return res
+
+
+class stock_picking_type(models.Model):
```
CamelCase for class name

pedrobaeza commented on stock_backorder_strategy/tests/test_backorder_strategy.py:
```diff
@@ -0,0 +1,81 @@
+
+class TestBackorderStrategy(common.TransactionCase):
+
+    def _process_picking(self, picking_type):
```
Why don't make this on setUp and then put each case on a isolated method? As it's rollbacked, you don't need to create 3 pickings, but just change the picking type backordering strategy

laetitia-gangloff commented:
@pedrobaeza Thank you very much !!

I adapt following your usefull comments.

pedrobaeza commented on stock_backorder_strategy/README.rst:
```diff
@@ -0,0 +1,66 @@
+Usage
+=====
+
+For further information, please visit:
```
You can also add the new cool button "Try me on runbot":

```
.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
   :alt: Try me on Runbot
   :target: https://runbot.odoo-community.org/runbot/154/8.0
```

pedrobaeza commented:
Thanks for the changes. One last minor remark, but :+1: when CIs goes green

yvaucher commented on stock_backorder_strategy/README.rst:
```diff
@@ -0,0 +1,66 @@
+In types of operation add a backorder strategy to be able to cancel created
+backorder or to avoid to create a backorder.
+
+Strategies possible are :
```
Would be better to write:

`Possible strategies are:`

yvaucher commented:
Just one nitpicking otherwise :+1:

Commit:
[ADD] add module stock_backorder_strategy

Define on stock_picking_type how to manage backorder.
create : default behavior
no_create : no backorder created
cancel : backorder is cancelled

laetitia-gangloff commented:
@yvaucher @pedrobaeza thank you :smile:

laetitia-gangloff commented:
Should I change the repository for stock-logistics-workflow (instead of stock-logistics-warehouse) ?

pedrobaeza commented:
Uhm, good question. Yeah, I think so, because pickings are transactional data that corresponds more to the workflow. A renaming to _stock_picking_backorder_strategy_ also seems reasonable.

laetitia-gangloff commented:
sorry for the mistake, I migrate this PR here : https://github.com/OCA/stock-logistics-workflow/pull/143
I close this one.
